homeassistant/helpers/area_registry.py

```diff
@@ -29,26 +28,25 @@ class EventAreaRegistryUpdatedData(TypedDict):
     area_id: str
 
 
-@attr.s(slots=True, frozen=True)
+@dataclasses.dataclass(frozen=True, kw_only=True, slots=True)
 class AreaEntry:
     """Area Registry Entry."""
 
-    name: str = attr.ib()
-    normalized_name: str = attr.ib()
-    aliases: set[str] = attr.ib(
-        converter=attr.converters.default_if_none(factory=set)  # type: ignore[misc]
-    )
-    id: str | None = attr.ib(default=None)
-    picture: str | None = attr.ib(default=None)
-
-    def generate_id(self, existing_ids: Container[str]) -> None:
-        """Initialize ID."""
-        suggestion = suggestion_base = slugify(self.name)
+    aliases: set[str]
+    id: str | None
+    name: str
+    normalized_name: str
+    picture: str | None
+
+    @staticmethod
+    def generate_id(existing_ids: Container[str], name: str) -> str:
```

Comment:
Can we use `__post_init__` with using [InitVar](https://docs.python.org/3/library/dataclasses.html#init-only-variables) for `existing_ids` instead?

I was thinking of something similar as I did for the `device_id` in https://github.com/DeebotUniverse/client.py/blob/e55e41b73e3c35ae34d3cc4dfaad0a31445e8f51/deebot_client/mqtt_client.py#L70

Reply:
Why do you use `__setattr__` instead of `self.device_id = "something"`?

Reply:
Note that it's not just any `__setattr__`, it's `object.__setattr__`, this is the documented way to bypass the overridden `__setattr__` in frozen dataclasses which otherwise prevents setting an attribute.

Reply:
I discussed it with Erik on Discord and `generate_id` can be outside the dataclass.
@MartinHjelmare Are you fine with the current implementation?